src/pages/Signup.tsx

```diff
@@ -0,0 +1,69 @@
+import Header from '@/components/common/Header';
+import ProgressBar from '@/components/signup/ProgressBar';
+import SignupButton from '@/components/signup/SignupButton';
+import { SIGNUP_PAGE_STYLE } from '@/constants/styles';
+import useSignupMutation from '@/hooks/useSignupMutation';
+import { useRef, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+
+const Signup = () => {
+  const nicknameRef = useRef<HTMLInputElement>(null);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const location = useLocation();
+  const { requestData, user } = location.state;
+
+  const onErrorCallback = (errorCode: string) => {
+    if (errorCode === '3401') {
+      setErrorMessage('중복된 닉네임입니다.');
+    } else {
+      setErrorMessage('회원가입 중 오류가 발생했습니다.');
+    }
+  };
+
+  const { mutateSignup, isPending } = useSignupMutation(onErrorCallback);
+
+  const handleSignup = async () => {
+    const nickname = nicknameRef.current?.value;
+
+    if (!nickname) {
+      setErrorMessage('닉네임을 입력해주세요.');
+      return;
+    }
+
+    setErrorMessage('');
+
+    user.nickname = nickname;
+    const param = { ...requestData, user };
+
+    mutateSignup(param);
+  };
+
+  return (
+    <div className={SIGNUP_PAGE_STYLE.constainer}>
+      <Header headerTitle="회원가입" />
+      <ProgressBar progress="half" />
+      <h2 className={SIGNUP_PAGE_STYLE.h2}>닉네임을 입력해주세요.</h2>
+      <div className={SIGNUP_PAGE_STYLE.form}>
+        <label htmlFor="nickname" className={SIGNUP_PAGE_STYLE.label}>
+          닉네임
+        </label>
+        <input
+          id="nickname"
+          type="text"
+          placeholder="닉네임을 입력해주세요."
+          className={SIGNUP_PAGE_STYLE.input}
+          ref={nicknameRef}
+        />
+        <p className={SIGNUP_PAGE_STYLE.errorMessage}>{errorMessage}</p>
+      </div>
+      <SignupButton
+        text={isPending ? '가입중...' : '가입하기'}
+        onClick={() => {
+          handleSignup();
```

Comment:
```suggestion
        onClick={() => {
          handleClick();
```
제가 좀 조사해본바로는 on[EventName]과 매칭되는게 handle[EvnetName]식으로, 처리하는걸로 알아봤습니다.
- https://stackoverflow.com/questions/60048249/what-is-the-right-name-of-event-handler-onclick-or-handleclick
꼭 지켜야되는지는 모르겠지만, 한번 고려해보는건 어떨까요?

Reply:
네 고려해 보겠습니다. 감사합니다.🙏🏻